src/Appwrite/Platform/Modules/Proxy/Http/Rules/Redirect/Create.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace Appwrite\Platform\Modules\Proxy\Http\Rules\Redirect;
+
+use Appwrite\Event\Certificate;
+use Appwrite\Event\Event;
+use Appwrite\Extend\Exception;
+use Appwrite\Network\Validator\CNAME;
+use Appwrite\SDK\AuthType;
+use Appwrite\SDK\Method;
+use Appwrite\SDK\Response as SDKResponse;
+use Appwrite\Utopia\Response;
+use Utopia\App;
+use Utopia\Database\Database;
+use Utopia\Database\Document;
+use Utopia\Database\Exception\Duplicate;
+use Utopia\Database\Helpers\ID;
+use Utopia\Domains\Domain;
+use Utopia\Platform\Action;
+use Utopia\Platform\Scope\HTTP;
+use Utopia\System\System;
+use Utopia\Validator\Domain as ValidatorDomain;
+
+class Create extends Action
+{
+    use HTTP;
+
+    public static function getName()
+    {
+        return 'createRedirectRule';
+    }
+
+    public function __construct()
+    {
+        $this
+            ->setHttpMethod(Action::HTTP_REQUEST_METHOD_POST)
+            ->setHttpPath('/v1/proxy/rules/redirect')
+            ->groups(['api', 'proxy'])
+            ->desc('Create Redirect rule')
+            ->label('scope', 'rules.write')
+            ->label('event', 'rules.[ruleId].create')
+            ->label('audits.event', 'rule.create')
+            ->label('audits.resource', 'rule/{response.$id}')
+            ->label('sdk', new Method(
+                namespace: 'proxy',
+                name: 'createRedirectRule',
+                description: <<<EOT
+                Create a new proxy rule for to redirect from custom domain to another domain.
+                EOT,
+                auth: [AuthType::ADMIN],
+                responses: [
+                    new SDKResponse(
+                        code: Response::STATUS_CODE_CREATED,
+                        model: Response::MODEL_PROXY_RULE,
+                    )
+                ]
+            ))
+            ->label('abuse-limit', 10)
+            ->label('abuse-key', 'userId:{userId}, url:{url}')
+            ->label('abuse-time', 60)
+            ->param('domain', null, new ValidatorDomain(), 'Domain name.')
+            ->param('target', null, new ValidatorDomain(), 'Target domain (hostname) of redirection')
+            ->inject('response')
+            ->inject('project')
+            ->inject('queueForCertificates')
+            ->inject('queueForEvents')
+            ->inject('dbForPlatform')
+            ->callback([$this, 'action']);
+    }
+
+    public function action(string $domain, string $target, Response $response, Document $project, Certificate $queueForCertificates, Event $queueForEvents, Database $dbForPlatform)
+    {
+        $mainDomain = System::getEnv('_APP_DOMAIN', '');
+        $sitesDomain = System::getEnv('_APP_DOMAIN_SITES', '');
+        $functionsDomain = System::getEnv('_APP_DOMAIN_FUNCTIONS', '');
+
+        $deniedDomains = [
+            $mainDomain,
+            $sitesDomain,
```

Comment:
Should sites and functions domains be in denied list for redirect rule?

Reply:
Yes, so you cannot "steal" any of thise domains, including "sites.localhost" which could be on Cloud "appwrite.sites"